gusto/time_discretisation.py

```diff
@@ -1341,3 +1465,40 @@ def apply(self, x_out, *x_in):
             self.x[n].assign(x_in[n])
         solver.solve()
         x_out.assign(self.x_out)
+
+
+class ImplicitMidpoint(ImplicitMultistage):
+    u"""
+    Implements the Implicit Midpoint method as a 1-stage Runge Kutta method.
+
+    The method, for solving
+    ∂y/∂t = F(y), can be written as:
+
+    k0 = F[y^n + 0.5*dt*k0]
+    y^(n+1) = y^n + dt*k0
+    """
+    def __init__(self, domain, field_name=None, solver_parameters=None, limiter=None, options=None, butcher_matrix=None):
+        super().__init__(domain, field_name,
+                         solver_parameters=solver_parameters,
+                         limiter=limiter, options=options)
+        self.butcher_matrix = np.array([[0.5], [1.]])
+        self.nStages = int(np.shape(self.butcher_matrix)[1])
+
+
+class QinZhang(ImplicitMultistage):
+    u"""
+    Implements Qin and Zhang's two-stage, 2nd order, implicit Runge–Kutta method.
+
+    The method, for solving
+    ∂y/∂t = F(y), can be written as:
+
+    k0 = F[y^n + 0.25*dt*k0]
+    k1 = F[y^n + 0.5*dt*k0 + 0.25*dt*k1]
+    y^(n+1) = y^n + 0.5*dt*(k0 + k1)
+    """
+    def __init__(self, domain, field_name=None, solver_parameters=None, limiter=None, options=None, butcher_matrix=None):
```

Comment:
Same here about:
- `butcher_matrix` not an argument
- `__init__` needs a docstring
- the summary docstring for the class should have linebreak characters included
- `butcher_matrix` should be passed through to `super().__init__`